src/compiler/types.ts

```diff
@@ -2893,6 +2901,61 @@ namespace ts {
         failedLookupLocations: string[];
     }
 
+    export type LintErrorMethod = {
+        (err: string): void;
+        (err: string, span: Node): void;
+        (err: string, start: number, length: number): void;
+    };
+    export type LintStopMethod = () => void;
+
+    /*
+    * Walkers call stop to halt recursion into the node's children
+    * Walkers call error to add errors to the output.
+    */
+    export interface LintWalker {
+        visit(node: Node, stop: LintStopMethod, error: LintErrorMethod): void;
+    }
+
+    export interface SyntacticLintProviderStatic {
+        new (typescript: typeof ts, args: any): LintWalker;
+    }
+
+    export interface SemanticLintProviderStatic {
+        new (typescript: typeof ts, checker: TypeChecker, args: any): LintWalker;
+    }
+
+    export namespace ExtensionKind {

```

Comment:
String literals don't get completions in value positions; namespace members do (and furthermore the namespace helps unify and organize them, much like enums). If you're suggesting we inline them, then maybe we should implement generic constant in-lining. ;)

Plus, access to them never happens on a particularly hot code path, so the perf difference should't matter.